modules/engines/engine-xssecurity/src/main/java/com/sap/xsk/xsaccess/ds/synchronizer/XSKSecuritySynchronizer.java

```diff
@@ -45,311 +44,301 @@
+     */
+    private void processArtifacts() throws SynchronizationException {
+        for (XSKPrivilegeDefinition xskPrivilegeDefinition : PRIVILEGES_TO_BE_PROCESSED) {
+            synchronizePrivilege(xskPrivilegeDefinition);
+        }
 
-      PRIVILEGES_SYNCHRONIZED.add(xskPrivilegeDefinition.getName());
-    } catch (XSKPrivilegeException e) {
-      throw new SynchronizationException(e);
+        for (XSKAccessDefinition xskAccessDefinition : ACCESS_TO_BE_PROCESSED) {
+            synchronizeAccess(xskAccessDefinition);
+        }
     }
-  }
-
-  /**
-   * Synchronize access.
-   *
-   * @param xskAccessDefinition the access definition
-   * @throws SynchronizationException the synchronization exception
-   */
-  private void synchronizeAccess(XSKAccessDefinition xskAccessDefinition) throws SynchronizationException {
-    try {
-      XSKAccessDefinition existingXSKAccessDefinition = xskAccessCoreService.getXSKAccessDefinition(xskAccessDefinition.getPath());
-      if (existingXSKAccessDefinition == null) {
-        xskAccessCoreService.createXSKAccessDefinition(xskAccessDefinition.getPath(), xskAccessDefinition.getAuthenticationMethodsAsList(),
-            xskAccessDefinition.getHash(),
-            xskAccessDefinition.isExposed(), xskAccessDefinition.getAuthorizationRolesAsList());
-        logger.info("Synchronized a new XSK Access definition [[{}]-[{}]] from location: {}",
-            xskAccessDefinition.getAuthenticationMethodsAsList(), String.join(",",
-                (xskAccessDefinition.getAuthorizationRolesAsList() != null ? xskAccessDefinition.getAuthorizationRolesAsList()
-                    : new ArrayList())), xskAccessDefinition.getPath());
-      } else if (!existingXSKAccessDefinition.getHash().equals(xskAccessDefinition.getHash())) {
-        xskAccessCoreService.updateXSKAccessDefinition(xskAccessDefinition.getPath(), xskAccessDefinition.getAuthenticationMethodsAsList(),
-            xskAccessDefinition.getHash(),
-            xskAccessDefinition.isExposed(), xskAccessDefinition.getAuthorizationRolesAsList());
-        logger.info("Synchronized a modified XSK Access definition [[{}]-[{}]] from location: {}",
-            xskAccessDefinition.getAuthenticationMethodsAsList(), String.join(",",
-                (xskAccessDefinition.getAuthorizationRolesAsList() != null ? xskAccessDefinition.getAuthorizationRolesAsList()
-                    : new ArrayList())), xskAccessDefinition.getPath());
-      }
-
-      ACCESS_SYNCHRONIZED.add(xskAccessDefinition.getPath());
-    } catch (XSKAccessException e) {
-      throw new SynchronizationException(e);
+
+    /**
+     * Synchronize privileges.
+     *
+     * @param xskPrivilegeDefinition the privilege definition
+     * @throws SynchronizationException the synchronization exception
+     */
+    private void synchronizePrivilege(XSKPrivilegeDefinition xskPrivilegeDefinition) throws SynchronizationException {
```

Comment:
Couldn't we refactor a bit this method not to have so many nested `if` statements?